### PR TITLE
HBASE-23055 Alter hbase:meta

### PR DESCRIPTION
**Pull request by saintstack**
Make hbase:meta region schema dynamic.

Patch has been under development a good while and its focus has changed
a few times so its bloated with fixup from older versions.

M hbase-server/src/main/java/org/apache/hadoop/hbase/master/TableStateManager.java
M hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/ZNodePaths.java
 Shut down access to internals and removed unused methods.

M hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/EnableTableProcedure.java
 Cleanup/refactor section on replica-handling.

M hbase-server/src/main/java/org/apache/hadoop/hbase/util/FSTableDescriptors.java
 Get hbase:meta schema from filesystem rather than from hard-coding.

### CODE REVIEW

**Comment by saintstack** [comment]
New PR to replace https://github.com/apache/hbase/pull/1011. Addresses comments left on #1011 .

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 6 new or modified test files.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 33s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 58s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   3m 32s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 13s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 59s |  branch-2 passed  |
| +0 :ok: |  spotbugs  |   0m 36s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 59s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 27s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  hbase-client: The patch generated 0 new + 83 unchanged - 4 fixed = 83 total (was 87)  |
| +1 :green_heart: |  checkstyle  |   0m 14s |  The patch passed checkstyle in hbase-zookeeper  |
| -1 :x: |  checkstyle  |   1m 26s |  hbase-server: The patch generated 1 new + 405 unchanged - 17 fixed = 406 total (was 422)  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  The patch passed checkstyle in hbase-shell  |
| +1 :green_heart: |  rubocop  |   0m  3s |  There were no new rubocop issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 11s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 25s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 55s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 33s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m  8s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 37s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 50s |  hbase-zookeeper in the patch passed.  |
| +1 :green_heart: |  unit  | 156m 19s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   7m 58s |  hbase-shell in the patch passed.  |
| +1 :green_heart: |  asflicense  |   2m 21s |  The patch does not generate ASF License warnings.  |
|  |   | 249m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1043 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile rubocop |
| uname | Linux 659d3340e582 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1043/out/precommit/personality/provided.sh |
| git revision | branch-2 / 4a35e2f837 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/1/testReport/ |
| Max. process+thread count | 4628 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-zookeeper hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.79.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
Fix checkstyle.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 40s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 6 new or modified test files.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 13s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 31s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   3m  1s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 13s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 51s |  branch-2 passed  |
| +0 :ok: |  spotbugs  |   0m 34s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 59s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 30s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  hbase-client: The patch generated 0 new + 83 unchanged - 4 fixed = 83 total (was 87)  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  The patch passed checkstyle in hbase-zookeeper  |
| -1 :x: |  checkstyle  |   1m 29s |  hbase-server: The patch generated 1 new + 405 unchanged - 17 fixed = 406 total (was 422)  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  The patch passed checkstyle in hbase-shell  |
| +1 :green_heart: |  rubocop  |   0m  3s |  There were no new rubocop issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 29s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 14s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 53s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m  5s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 44s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 53s |  hbase-zookeeper in the patch passed.  |
| -1 :x: |  unit  | 169m 38s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  unit  |   8m 32s |  hbase-shell in the patch passed.  |
| +1 :green_heart: |  asflicense  |   2m 12s |  The patch does not generate ASF License warnings.  |
|  |   | 259m 40s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.TestHBaseMetaEdit |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1043 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile rubocop |
| uname | Linux 21bdd08febb1 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1043/out/precommit/personality/provided.sh |
| git revision | branch-2 / ae95b1f215 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/2/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/2/testReport/ |
| Max. process+thread count | 5121 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-zookeeper hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.79.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 6 new or modified test files.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 15s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 31s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   3m  5s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 29s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 56s |  branch-2 passed  |
| +0 :ok: |  spotbugs  |   0m 39s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   6m 38s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 22s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 33s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 33s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  hbase-client: The patch generated 0 new + 83 unchanged - 4 fixed = 83 total (was 87)  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  The patch passed checkstyle in hbase-zookeeper  |
| -1 :x: |  checkstyle  |   1m 29s |  hbase-server: The patch generated 1 new + 405 unchanged - 17 fixed = 406 total (was 422)  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  The patch passed checkstyle in hbase-shell  |
| +1 :green_heart: |  rubocop  |   0m  3s |  There were no new rubocop issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 18s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 24s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 51s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m  8s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 38s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 53s |  hbase-zookeeper in the patch passed.  |
| +1 :green_heart: |  unit  | 149m 56s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 48s |  hbase-shell in the patch passed.  |
| +1 :green_heart: |  asflicense  |   2m 43s |  The patch does not generate ASF License warnings.  |
|  |   | 240m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1043 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile rubocop |
| uname | Linux 943280315d73 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1043/out/precommit/personality/provided.sh |
| git revision | branch-2 / 31b4fbdee5 |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/3/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/3/testReport/ |
| Max. process+thread count | 5177 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-zookeeper hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.79.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
Fix checkstyle. Any chance of review? Thanks.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 6 new or modified test files.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 19s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 31s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   3m  3s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 18s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 55s |  branch-2 passed  |
| +0 :ok: |  spotbugs  |   0m 36s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   6m  5s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 32s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 32s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  The patch passed checkstyle in hbase-common  |
| +1 :green_heart: |  checkstyle  |   0m 38s |  hbase-client: The patch generated 0 new + 83 unchanged - 4 fixed = 83 total (was 87)  |
| +1 :green_heart: |  checkstyle  |   0m 14s |  The patch passed checkstyle in hbase-zookeeper  |
| +1 :green_heart: |  checkstyle  |   1m 29s |  hbase-server: The patch generated 0 new + 404 unchanged - 17 fixed = 404 total (was 421)  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  The patch passed checkstyle in hbase-shell  |
| +1 :green_heart: |  rubocop  |   0m  3s |  There were no new rubocop issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 12s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 22s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   1m 55s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   6m 30s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 10s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 42s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 53s |  hbase-zookeeper in the patch passed.  |
| +1 :green_heart: |  unit  | 150m 45s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |   9m  0s |  hbase-shell in the patch passed.  |
| +1 :green_heart: |  asflicense  |   2m 45s |  The patch does not generate ASF License warnings.  |
|  |   | 240m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1043 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile rubocop |
| uname | Linux 9ff8e0a84da1 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1043/out/precommit/personality/provided.sh |
| git revision | branch-2 / 1ea389a834 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/4/testReport/ |
| Max. process+thread count | 5024 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-zookeeper hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1043/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.79.0 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
WIll push this in next day or so unless objection. Review appreciated.

**Comment by saintstack** [comment]
Merged. 3rd time lucky I hope.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ModifyTableProcedure.java [hunk 1]
```diff
@@ -59,10 +62,19 @@
   private TableDescriptor modifiedTableDescriptor;
   private boolean deleteColumnFamilyInModify;
   private boolean shouldCheckDescriptor;
+  /**
+   * List of column families that cannot be deleted from the hbase:meta table.
+   * They are critical to cluster operation. This is a bit of an odd place to
+   * keep this list but then this is the tooling that does add/remove. Keeping
+   * it local!
+   */
+  private static final List<byte []> UNDELETABLE_META_COLUMNFAMILIES =
```
Comment (Apache9) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ModifyTableProcedure.java]: There are other things we can not change. For example we should not set TTL on the columnf families, and also, you should not change the versions of REPLICATION_BARRIER_FAMILY...

Can be a follow on.

Comment (saintstack) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ModifyTableProcedure.java]: Filed HBASE-23721

**Pull request merged**